gradle-conjure/src/main/java/com/palantir/gradle/conjure/CompileConjureTypeScriptTask.java

```diff
@@ -32,16 +33,12 @@
 import org.gradle.api.tasks.PathSensitivity;
 
 @CacheableTask
-public class CompileConjureTypeScriptTask extends ConjureGeneratorTask {
-    private final Property<String> packageName = getProject().getObjects().property(String.class);
-    private final Property<String> packageVersion = getProject().getObjects().property(String.class);
-
-    private File productDependencyFile;
-
+public abstract class CompileConjureTypeScriptTask extends ConjureGeneratorTask {
     public CompileConjureTypeScriptTask() {
         Project project = getProject();
-        packageName.set(getProject().provider(project::getName));
-        packageVersion.set(getProject().provider(() -> project.getVersion().toString()));
+        getPackageName().convention(getProject().provider(project::getName));
```

Comment:
I'm wondering what the difference between `property.convention` and `property.value/set` do.  Both seem to set default values, but I haven't actually seen `convention` used until just now. I thought they might be name overloads but they do seem to have different behaviour in the code.

Reply:
Ah, from https://docs.gradle.org/7.2/userguide/lazy_configuration.html#applying_conventions:

> Once a value is set, the convention is ignored

So this means if you set a convention after there is a value, it's a no-op, where as `value`/`set` would actually replace the value. Sounds like a useful tool - in this case sense we're calling `convention` right after making the properties it should be equivalent to `value`/`set`.

Reply:
yes, convention is setting a default value so you can tell if it's explicitly overridden with a set call.  I have always pictured that it would be nice to develop a way where you can collect metrics on if a property is every overridden.  That way you can tell if changing/removing the property would cause a break to consumers.  Just never did anything with the idea.